components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayJMSMessageListener.java

```diff
@@ -291,4 +297,35 @@ private void startTenantFlow(String tenantDomain) {
         PrivilegedCarbonContext.getThreadLocalCarbonContext().
                 setTenantDomain(tenantDomain, true);
     }
+
+    private synchronized void handleAsyncWebhooksSubscriptionMessage(JsonNode payloadData) {
```

Comment:
do we need to synchronize this method? addSubscription() method inserts data to a concurrent hash map right?